### PR TITLE
Re-exposes dataObjInp_t to Native Rule Language - (master)

### PR DESCRIPTION
**Pull request by korydraughn**
Tested in Docker.

All tests passed in Docker except the following (which I believe are docker related issues):
- test_resource_types
- test_resource_tree

### CODE REVIEW

**Comment by trel** [comment]
looks good to me - the RODS_CLERVER dance had only been added in master?

**Comment by korydraughn** [comment]
Yes. Other discussions about this macro's purpose happened which lead to this commit.

File: server/re/src/irods_re_serialization.cpp [hunk 1]
```diff
@@ -551,6 +551,32 @@ namespace irods {
...
+                }
+            }
+            catch ( std::exception& ) {
+                return ERROR(
+                         INVALID_ANY_CAST,
+                         "failed to cast collInp ptr" );
+            }
+
+            return SUCCESS();
+        } // serialize_collInfo_ptr
```
Comment (alanking) [inline server/re/src/irods_re_serialization.cpp]: Nit: serialize_collInp_ptr

**Comment by korydraughn** [comment]
Added #

**Pull request merged**